File: Python/Array/Minimum swaps and K together.py
def minSwap (arr, n, k) : 
    
    count = 0
    for i in range(0, n) :
        if (arr[i] <= k) :
            count = count + 1
     
    bad = 0
    for i in range(0, count) :
        if (arr[i] > k) :
            bad = bad + 1
     
    ans = bad
    j = count
    for i in range(0, n) :
         
        if(j == n) :
            break
             
        if (arr[i] > k) :
            bad = bad - 1
         
        if (arr[j] > k) :
            bad = bad + 1
         
        ans = min(ans, bad)
 
        j = j + 1
 
    return ans

File: C++/Array/Minimum swaps and K together.cpp

int minSwap(int *arr, int n, int k) {
    // Complet the function
    int count = 0;
    for (int i = 0; i < n; ++i)
        if (arr[i] <= k)
            ++count;
     
    int bad = 0;
    for (int i = 0; i < count; ++i)
        if (arr[i] > k)
            ++bad;
     
    int ans = bad;
    for (int i = 0, j = count; j < n; ++i, ++j) {
         
        if (arr[i] > k)
            --bad;
         
        if (arr[j] > k)
            ++bad;
         
        ans = min(ans, bad);
    }
    return ans;
}
